source/agora/test/Base.d

```diff
@@ -200,9 +200,16 @@ public NetworkT makeTestNetwork (NetworkT : TestNetwork)
 
         auto net = new NetworkT(NodeConfig.init, node_configs.map!(
             c => c.key_pair.address.toString()).array);
+
+        // accessibility workaround: Base module has access to protected
+        // createNewNode because it's defined in TestNetwork in *this* module,
+        // however it cannot access NetworkT's override because it usually
+        // lives in another module
+        auto base_net = cast(TestNetwork)net;
```

Comment:
Can you report the DMD bug ?
Then just:
```d
auto ret = ....
// Workaround LINK-TO-BUGZILLA
TestNetwork net = ret;
[...]
return ret;
```

I don't like `cast` because it is not future proof: If you change the type, an implicit conversion will fail, but a `cast` will just return `null`.

Reply:
I have a feeling that it's already reported somewhere. I'll try to find it.

Reply:
And yeah I shouldn't cast.

Reply:
It seems it's similar to https://issues.dlang.org/show_bug.cgi?id=2417, but there they just changed the spec as a "wontfix". (even though it says "fixed")

Reply:
However I will file a new one.

Reply:
Filed as https://issues.dlang.org/show_bug.cgi?id=20002 and updated the PR.